plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -405,12 +405,34 @@ private function process_select_field( $field_data, $options ) {
 	private function process_checkbox_field( $field_data, $options ) {
 		$id = $options['id'];
 
-		// Checkbox fields are always optional. Log a warning if it's set explicitly as true.
-		$field_data['required'] = false;
+		if ( isset( $options['required'] ) && null === filter_var( $options['required'], FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE ) ) {
+			$message = sprintf( 'The required property for field with id: "%s" must be a boolean, you passed %s. The field will not be registered.', $id, gettype( $options['required'] ) );
+			_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), '9.8.0' );
+			return false;
+		}
 
-		if ( isset( $options['required'] ) && true === $options['required'] ) {
-			$message = sprintf( 'Registering checkbox fields as required is not supported. "%s" will be registered as optional.', $id );
-			_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), '8.6.0' );
+		if ( isset( $options['required'] ) ) {
+			// Filter the boolean in case the developer added it as a string.
+			$options['required']    = filter_var( $options['required'], FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE );
+			$field_data['required'] = $options['required'];
+		}
+
```

Comment:
This wouldn't be needed anymore. At this point `$field_data['required']` === `true`

```suggestion

```

Reply:
I'm going to remove setting `$options['required']` and instead set it straight to `$field_data['required']` I don't think we need to reference `$options` again and can just use `$field_data` from here.